### PR TITLE
Fix reading special character issue for tool resolver

### PR DESCRIPTION
**Pull request by huaiyan**
# Description

For LLM/Prompt/CUSTOM_LLM tool, we need to read file to resolve the tool.
If there is special character, will introduce unexpected error.
So, we set encoding as "uts-8".

This is continuing PR for https://github.com/microsoft/promptflow/pull/759

# All Promptflow Contribution checklist:
- [X] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [X] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [X] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/executor/_tool_resolver.py [hunk 1]
```diff
@@ -129,15 +129,16 @@ def resolve_tool_by_node(self, node: Node, convert_input_types=True) -> Resolved
...
+        if source is None or source.path is None or not Path(self._working_dir / source.path).is_file():
             raise InvalidSource(
                 target=ErrorTarget.EXECUTOR,
                 message_format="Node source path '{source_path}' is invalid on node '{node_name}'.",
                 source_path=source.path if source is not None else None,
                 node_name=node.name,
             )
-        with open(self._working_dir / source.path) as fin:
-            return fin.read()
+        file = Path(self._working_dir / source.path)
```
Comment (zzn2) [inline src/promptflow/promptflow/executor/_tool_resolver.py:140]: Can be simplified to `file = self._working_dir / source.path`? As long as the `/` operator works, we can confirm that `self._working_dir` is already a `Path` object.

File: src/promptflow/tests/test_configs/flows/script_with_special_character/script_with_special_character.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from promptflow import tool
+
+@tool
+def print_special_character(input1: str) -> str:
+    # Add special character to test if file read is working.
+    return "https://www.bing.com//"
```
Comment (zzn2) [inline src/promptflow/tests/test_configs/flows/script_with_special_character/script_with_special_character.py:6]: Can we use `\ue000\ue001` here too to align with the assertion code?
```python
        assert "https://www.bing.com/\ue000\ue001/" in result
```

Comment (huaiyan) [reply src/promptflow/tests/test_configs/flows/script_with_special_character/script_with_special_character.py:6]: When we use `\ue000\ue001`, the result will be `\\ue000\\ue001`

Comment (zzn2) [reply src/promptflow/tests/test_configs/flows/script_with_special_character/script_with_special_character.py:6]: I see.

**Review by zzn2 — DISMISSED** [review]

**Review by lumoslnt — APPROVED** [review]

**Review by lumoslnt — APPROVED** [review]

**Pull request merged**